### PR TITLE
Updated to use latest version of ZAPD

### PR DESCRIPTION
**Pull request by NEstelami**

### CODE REVIEW

File: extract_assets.py
Comment (AngheloAlf) [inline extract_assets.py:15]: `./extract_assets.py -s` doesn't work anymore because of this. Please fix

File: extract_assets.py
Comment (AngheloAlf) [inline extract_assets.py:22]: Remove commented code.

File: tools/ZAPDConfigs/MqDbg/Config.xml
Comment (AngheloAlf) [inline tools/ZAPDConfigs/MqDbg/Config.xml:19]: Is there any reason to have those commented lines here?

**Review by AngheloAlf — CHANGES_REQUESTED** [review]
Because of some changes in ZAPD, there are over 700 new newline warnings, which are already fixed in the current OoT master branch.
The background (jpeg files) extraction is broken too. It just doesn't extract the jpgs as it should.

[PR 107](https://github.com/zeldaret/ZAPD/pull/107) in ZAPD's repo fixes those two issues, I suggest merging that first.

Don't forget to fix `extract_assets.py` too.

**Comment by NEstelami** [comment]
> The background (jpeg files) extraction is broken too. It just doesn't extract the jpgs as it should.
> 
> [PR 107](https://github.com/zeldaret/ZAPD/pull/107) in ZAPD's repo fixes those two issues, I suggest merging that first.

JPEG extraction is still working fine on my end, but I'll merge in your PR.

File: tools/ZAPDConfigs/MqDbg/Config.xml
Comment (NEstelami) [reply tools/ZAPDConfigs/MqDbg/Config.xml:19]: That was meant to be removed.

**Review by AngheloAlf — APPROVED** [review]

**Comment by Roman971** [comment]
Could you describe the changes included in these changes, from a functional standpoint? Some of them might need a discussion so it would be good to have a clear list of changes

**Comment by fig02** [comment]
Given that none of the xmls changed, I wanna say that nothing functional has really changed. Just a big refactor. Could be wrong tho, Jack should confirm

**Comment by NEstelami** [comment]
The main things that have changed as far as features go are the introduction of a config file and the splitting of vertices into their own file. The config file will allow for multiple games/game versions to be supported without modification of ZAP's source code. Right now that primarily covers actor and object lists, but it'll allow for more as ZAP gets updated.

**Comment by AngheloAlf** [comment]
Changes:
- Throw an exception if "Height" or "Width" is not spelled correctly. ([104](https://github.com/zeldaret/ZAPD/pull/104))
- Unmangle backtrace. ([100](https://github.com/zeldaret/ZAPD/pull/100))
  - When ZAPD crashes, the backtrace will be something readable.
- Add backgrounds (`.jpg`) extraction. ([106](https://github.com/zeldaret/ZAPD/pull/106) and [107](https://github.com/zeldaret/ZAPD/pull/107))
  - Use a `<Background>` tag in xml or let ZAPD extract every background it is able to find.
- Add Mtx extraction. ([105](https://github.com/zeldaret/ZAPD/pull/105))
  - `<Mtx>`. It also auto-extracts the ones used in DLists.
  - Related PR: #603 
- Add MM scene support. ([109](https://github.com/zeldaret/ZAPD/pull/109))
- Big code refactor. ([108](https://github.com/zeldaret/ZAPD/pull/108) and [112](https://github.com/zeldaret/ZAPD/pull/112))
  - Config file.
  - Add a parameter to use the legacy display list disassembly.
  - Refactor ZFile.
  - Improve how `Vtx`s are extracted.
  - Vertex arrays are stored in .inc.c files now.
  - WIP improvements to error handling.

I probably forgot to mention something. If anybody is curious, you can check the commit list [here](https://github.com/zeldaret/ZAPD/compare/c4773301a17a5a9f9c964a3e7ba6dfbf0730a94b...179af7d11fd27b046edfb1bbadade865033583d7).

**Review by fig02 — APPROVED** [review]

**Comment by fig02** [comment]
I figure we can wait to get https://github.com/zeldaret/ZAPD/pull/117
and possibly https://github.com/zeldaret/ZAPD/pull/115
in before merging this

**Comment by NEstelami** [comment]
> I figure we can wait to get [zeldaret/ZAPD#117](https://github.com/zeldaret/ZAPD/pull/117)
> and possibly [zeldaret/ZAPD#115](https://github.com/zeldaret/ZAPD/pull/115)
> in before merging this

They've been merged. Is anything else needed?

**Comment by louist103** [comment]
122 might be nice to have.

**Comment by fig02** [comment]
Yea, output formatter would be pretty good to bring in. But if that needs more time to review, we can bring this in and make it a separate pr later

File: Makefile
Comment (louist103) [inline Makefile:237]: Is there a reason for this change?

Comment (NEstelami) [reply Makefile:237]: I think that was done by accident or as part of some test I didn't revert afterwards.

Comment (louist103) [reply Makefile:237]: TBH I might have been the one who did it but thanks for fixing it for me.

**Comment by AngheloAlf** [comment]
ZAPD changes:
- Changes described [here](https://github.com/zeldaret/oot/pull/777#issuecomment-821863019).
- Code house keeping. Just a general commented code cleanup. ([114](https://github.com/zeldaret/ZAPD/pull/114))
- ZAPD extraction xml reference. A small reference/documentation for the XML node/elements for assets extraction. ([102](https://github.com/zeldaret/ZAPD/pull/102))
- Deprecate `Hint` system. ([116](https://github.com/zeldaret/ZAPD/pull/116))
  - The `Hint` system will be removed in the future, this is to provide an easier migration for existing XMLs which already uses some hints.
  - Remove `BlobHint`, `TextureHint` and `BackgroundHint`.
  - Add a "deprecation warning". It will trigger if any of the "Hint" nodes are found in a XML file.
  - Fix the `Cutscene` node, so it can be used instead of `CutsceneHint`.
- Name small zero-only unaccounted as `possiblePadding_` and optional unaccounted warning. ([118](https://github.com/zeldaret/ZAPD/pull/118)) 
  - Rename small zero-only unaccounted blocks.
    - If the size of an unaccounted block is less than `0x10` bytes and only has zeroes, then it will be named `possiblePadding_` instead of `unaccounted_`.
  - Add two optional warnings.
    - Warn when a block of unaccounted (non-zero) data of any size is found (`A non-zero unaccounted block was found at address`). Can be useful when decompiling objects.
    - Warn when an unaccounted block of zero-only data is found (`A big (size>=0x10) zero-only unaccounted block was found at address`). This warning is triggered only if the block's size is at least `0x10` bytes.
    - This warnings are disabled by default. To enable them, pass the flag `-wu` (or `--warn-unaccounted`) to ZAPD.
- OutputFormatter fixes. ([117](https://github.com/zeldaret/ZAPD/pull/117))
  - Fixes some weird formatting issues that were observed in assets/overlays/ovl_Boss_Sst/ovl_Boss_Sst.c of the oot repo.
- Add a few errors checks. ([119](https://github.com/zeldaret/ZAPD/pull/119))
  - Check if an unknown resource was declared in a XML file.
  - Check for repeated names, outnames and offsets.
  - Check for valid variable name for each resource.
  - Among others.
- Fix a small padding problem in ZCutsceneMM. ([120](https://github.com/zeldaret/ZAPD/pull/120))
- Moved BG Screen Width and Height into Config File [121](https://github.com/zeldaret/ZAPD/pull/121)
  - Allows to define the dimensions of the screen on the ZAPD config file.
- Fix most `-Wall` warnings. ([115](https://github.com/zeldaret/ZAPD/pull/115))
- Texture Pool Fully Implemented. ([123](https://github.com/zeldaret/ZAPD/pull/123))
  - Keep track of duplicated textures and re use them instead of re-extracting them.

I hope I didn't forget something. For the commit list, see [here](https://github.com/zeldaret/ZAPD/compare/179af7d11fd27b046edfb1bbadade865033583d7...50ad2fe786af4cf66998028a08c767e16a416a60).

~How is this list larger than the first one?~

**Comment by fig02** [comment]
if you wanna do a subrepo pull to bring in output formatter, I think this is good to go after that

**Pull request merged**
